packages/interface/src/index.ts

```diff
@@ -0,0 +1,61 @@
+type Mnemonic = string
+type Address = string
+enum Chain {
+  BITCOIN,
+  ETHEREUM,
+  NEAR,
+  TERRA,
+  SOLANA
+}
+
+enum Network {
+  MAINNET,
+  TESTNET
+}
+
+enum Token {
+  NATIVE,
+  ERC20,
+  ERC721
+}
+
+enum Hardware {
+  LEDGER
+}
+
+export default interface IWalletConstructor {
+  new (mnemomic: Mnemonic, network: Network): IWallet
+}
+
+export interface IWallet {
```

Comment:
Do we want this to reflect the schema of the state object that we are currently using in the extension? Please take a look at it to see what I mean.

Reply:
Also, how would this be used with the the existing interfaces/classes (WalletManagerI, StorageManagerI, EncryptionManagerI)?